lunamark/reader/markdown.lua

```diff
@@ -1312,27 +1290,14 @@ function M.new(writer, options)
                                                      writer.string(infostring))
                          end
 
-
-  larsers.FencedDivBegin = larsers.fenced_div_begin
-                         * Cmt(P(true), function(_, i)
-                             -- keep track how deep nested we are
-                             div_level = div_level + 1
-                             return i
-                           end)
-
-  larsers.FencedDivEnd   = larsers.fenced_div_end
-                         * Cmt(P(true), function(_, i)
-                             local result
-                             -- check whether we are at top level and decrement
-                             -- div_level otherwise
-                             if div_level <= 0 then
-                               result = nil
-                             else
-                               div_level = div_level - 1
-                               result = i
-                             end
-                             return result
-                           end)
+    larsers.FencedDiv = P{ (larsers.fenced_div_begin)
+                           * C(((parsers.any - (larsers.fenced_div_begin + larsers.fenced_div_end)) + V(1))^0)
+                           * larsers.fenced_div_end } / function (attr, div)
+                                                          -- The \n here are to force paragraphs in the ..div
+                                                          -- Not sure Pandoc would generate Para nodes, rather than
+                                                          -- Plain, though!
+                                                          return parse_blocks("\n"..div.."\n"), attr
```

Comment:
It would. That's what https://github.com/jgm/lunamark/pull/51/commits/6dbd9244130e58e90de4596fda4ed57cf895ddfa in #51 was for.